analyzer/src/main/kotlin/managers/Bower.kt

```diff
@@ -141,13 +141,69 @@ class Bower(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfigu
+            // however in particular for root projects the null case also happens.
+            val name = node["pkgMeta"]["name"].textValueOrEmpty()
+            val version = node["pkgMeta"]["version"].textValueOrEmpty()
+            return if (name.isEmpty() || version.isEmpty()) null else "$name:$version"
+        }
+
+        private fun getNodesWithCompleteDependencies(node: JsonNode): Map<String, JsonNode> {
+            val result = mutableMapOf<String, JsonNode>()
+
+            val stack = Stack<JsonNode>()
+            stack.push(node)
+            while (!stack.empty()) {
+                val currentNode = stack.pop()
+
+                val key = dependencyKeyOf(currentNode)
+                if (key != null && hasCompleteDependencies(node, SCOPE_NAME_DEPENDENCIES)
+                        && hasCompleteDependencies(node, SCOPE_NAME_DEV_DEPENDENCIES)) {
+                    result[key] = currentNode
+                }
+
+                stack.addAll(getDependencyNodes(currentNode))
+            }
+
+            return result
+        }
+
         private fun extractDependencyTree(node: JsonNode, scopeName: String): SortedSet<PackageReference> {
+            return extractDependencyTree(node, scopeName, getNodesWithCompleteDependencies(node))
+        }
+
+        private fun extractDependencyTree(node: JsonNode, scopeName: String, alternativeNodes: Map<String, JsonNode>):
+                SortedSet<PackageReference> {
             val result = mutableSetOf<PackageReference>()
 
+            if (!hasCompleteDependencies(node, scopeName)) {
+                // Bower leaves out a dependency entry for a child if there exists a similar node to its parent node
+                // with the exact same name and resolved target. This makes it necessary to retrieve the information
+                // about the subtree rootet at the parent from that other node containing the full dependency
+                // information.
+                // See https://github.com/bower/bower/blob/6bc778d/lib/core/Manager.js:557 and below.
+
+                val alternativeNode = alternativeNodes[dependencyKeyOf(node)]
+                if (alternativeNode != null) {
+                    return extractDependencyTree(alternativeNode, scopeName, alternativeNodes)
+                } else {
+                    // This should never happen assuming 'bower list' exited with success.
+                }
+            }
+
             node["pkgMeta"][scopeName].fieldNamesOrEmpty().forEach {
-                val childNode = node["dependencies"][it]
+                var childNode = node["dependencies"][it]
                 val childScope = SCOPE_NAME_DEPENDENCIES
-                val childDependencies = extractDependencyTree(childNode, childScope)
+                val childDependencies = extractDependencyTree(childNode, childScope, alternativeNodes)
                 val packageRefence = PackageReference(
```

Comment:
typo: "packageReference"

Reply:
Right, but as that's in untouched code, I've fixed this separately in https://github.com/heremaps/oss-review-toolkit/pull/912.